### PR TITLE
Simplify the "ReaderHeadersReady" message-handler in the API

### PR DESCRIPTION
**Pull request by Snuffleupagus**
We can convert the handler to an `async` function, which removes the need to create a temporary Promise here.
Given the age of this code it shouldn't hurt to simplify it a little bit.

### CODE REVIEW

File: src/display/api.js [hunk 1]
```diff
@@ -2647,12 +2647,15 @@ class WorkerTransport {
     });
 
     messageHandler.on("ReaderHeadersReady", data => {
```
Comment (calixteman) [inline src/display/api.js:2649]: Can't we just male this function `async` and use `await` in its body ?

**Review by calixteman — APPROVED** [review]
LGTM. Thank you.

File: src/display/api.js [hunk 2]
```diff
@@ -2646,32 +2646,27 @@ class WorkerTransport {
...
-          }
-          fullReader.onProgress = evt => {
-            loadingTask.onProgress?.({
-              loaded: evt.loaded,
-              total: evt.total,
-            });
-          };
-        }
+    messageHandler.on("ReaderHeadersReady", async data => {
+      await this._fullReader.headersReady;
```
Comment (calixteman) [inline src/display/api.js:2650]: Just an idea: could we just have a getter/function in order to have something like `const { isStreamingSupported, ...} = await this._fullReader.FooBar()` ?

Comment (Snuffleupagus) [reply src/display/api.js:2650]: That'd be possible, however after considering it I'd prefer not to given the amount of code duplication that it'd lead to. Please note that we have no less than *four* different `IPDFStreamReader` implementations:
 - https://github.com/mozilla/pdf.js/blob/master/src/display/fetch_stream.js
 - https://github.com/mozilla/pdf.js/blob/master/src/display/network.js
 - https://github.com/mozilla/pdf.js/blob/master/src/display/node_stream.js
 - https://github.com/mozilla/pdf.js/blob/master/src/display/transport_stream.js

**Pull request merged**